src/redteam/strategies/retry.ts

```diff
@@ -1,11 +1,21 @@
 import { and, eq, sql } from 'drizzle-orm';
 import { getDb } from '../../database/index';
 import { evalResultsTable } from '../../database/tables';
+import { getEnvString } from '../../envars';
+import { cloudConfig } from '../../globalConfig/cloud';
 import logger from '../../logger';
+import { makeRequest } from '../../util/cloud';
 import invariant from '../../util/invariant';
 
 import type { TestCase, TestCaseWithPlugin } from '../../types/index';
 
+/**
+ * Check if we should use cloud mode for fetching failed test cases
+ */
+function isCloudMode(): boolean {
+  return !!getEnvString('PROMPTFOO_REMOTE_GENERATION_URL') || cloudConfig.isEnabled();
```

Comment:
it's ok to only check  cloudConfig.isEnabled() here - or consider centralizing this function